src/pages/order-history/order-history.js

```diff
@@ -1,40 +1,47 @@
-import React, { useEffect } from 'react';
-import { useDispatch, useSelector } from 'react-redux';
+import React, { useState } from 'react';
+import { useQuery } from '@apollo/client';
 import { useTranslation } from 'react-i18next';
-import { getUserOrders, setCurrentPage } from '../../redux/user/user.actions';
 import { Loader } from '../../components/loader/loader';
 import OrderHistoryOrder from '../../containers/orders/order-history/order-history-order';
 import EmptyOrderHistory from '../../containers/orders/order-history/empty-order-history';
 import OrderHistoryPagination from '../../containers/orders/order-history/order-history-pagination/index';
 import { useStyles } from './order-history.styles';
 import { limitHistoryOrders } from '../../const/user-order-history';
+import errorOrLoadingHandler from '../../utils/errorOrLoadingHandler';
+import { getUserOrdersCountQuery, getUserOrdersQuery } from './operations/order-history.queries';
 
 const OrderHistory = () => {
-  const { orders, loading, currentPage, countPerPage } = useSelector(({ User }) => ({
-    orders: User.userOrders,
-    loading: User.userLoading,
-    currentPage: User.currentPage,
-    countPerPage: User.countPerPage
-  }));
+  const [currentPage, setCurrentPage] = useState(1);
+  const [countPerPage, setCountPerPage] = useState(null);
+  const [orders, setOrders] = useState([]);
   const { t } = useTranslation();
-  const dispatch = useDispatch();
   const styles = useStyles();
 
-  const quantityPages = Math.ceil(countPerPage / limitHistoryOrders);
+  const { loading, errorOrders } = useQuery(getUserOrdersQuery, {
+    variables: {
+      pagination: {
+        limit: limitHistoryOrders,
+        skip: (currentPage - 1) * limitHistoryOrders
+      }
+    },
+    onCompleted: (data) => {
+      setOrders(data.getUserOrders);
+    }
+  });
 
-  useEffect(() => {
-    dispatch(
-      getUserOrders({
-        pagination: {
-          limit: limitHistoryOrders,
-          skip: (currentPage - 1) * limitHistoryOrders
-        }
-      })
-    );
-  }, [currentPage, countPerPage]);
+  const { loadingCount, errorCount } = useQuery(getUserOrdersCountQuery, {
+    variables: {
+      id: '617036ea0f1341197c925321'
```

Comment:
id: '617036ea0f1341197c925321' - So all the users will use same id ?

Reply:
my bad, will fix it a bit later